src/lib/optimizer/strategy/abstract_rule.hpp

```diff
@@ -2,34 +2,54 @@
 
 #include <memory>
 #include <string>
+#include <unordered_set>
+#include <vector>
 
 namespace opossum {
 
-class AbstractLQPNode;
-class AbstractCostEstimator;
 class AbstractCardinalityEstimator;
+class AbstractCostEstimator;
+class AbstractLQPNode;
+class LQPSubqueryExpression;
 
 class AbstractRule {
  public:
   virtual ~AbstractRule() = default;
 
   /**
    * This function applies the concrete Optimizer Rule to an LQP.
-   * apply_to() is intended to be called recursively by the concrete rule.
-   * The optimizer will pass the immutable LogicalPlanRootNode to this function.
+   * The optimizer will pass the immutable LogicalPlanRootNode (@param lqp_root) to this function.
```

Comment:
Is this comment really correct? The `AbstractLQPNode` is not const.

Reply:
The LogicalPlanRootNode stays, no matter which rules are applied. 
Removing `immutable` from the Doc is not critical though.